### PR TITLE
[luci] Introduce cleanup of unused inputs/outputs

### PR DESCRIPTION
**Pull request by seanshpark**
This will introduce remove_unused_inputoutputs method that removes
unused inputs/outputs in all PGroup of PGroups.

ONE-DCO-1.0-Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
For #6230, from draft #6231

File: compiler/luci/partition/src/PartitionCleanup.cpp [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  for (auto &pgroup : pgroups->pgroups)
+  {
+    bool changed;
+    do
+    {
+      changed = false;
+      for (auto it = pgroup->inputs.begin(); it != pgroup->inputs.end(); ++it)
+      {
+        auto input = *it;
+        if (pgroups->pgroup_of(input) == pgroup.get())
```
Comment (mhs4670go) [inline compiler/luci/partition/src/PartitionCleanup.cpp:89]: It's a little bit hard to understand. 

`input` is from `pgroup`. So, `pgroups_of(input)` returns `pgroup`. Then, `== pgroup.get()` condition seems always true. Anything wrong?

Comment (seanshpark) [reply compiler/luci/partition/src/PartitionCleanup.cpp:89]: `pgroup_of(input)` returns `pgroup` pointer of `input` which is `CircleNode`.

here, `input` is input of `pgroup` which may be in another `pgroup`
when `pgroup` is created, there is one `pnode` having `node`
the `input` is the input of this `node`, which will be in another `pgroup`

when two `pgroup` merges, first all the `input` are copied.
and remove not effective `input`.

this line is to check if the `input` is pointing to the merged `pgroup`.
before merge `pgroup` may have been another object.
if `pgroup` of `input` is same as this `pgroup` then we can think that
input is pointing to itself so we can remove this `input` from the `inputs` list.

**Review by mhs4670go — APPROVED** [review]
LGTM

**Pull request merged**